PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['mail.template']
-    _inherit = ['res.config.settings']
-    _inherit = ['res.partner']

Total Changes: 14

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/event/models/event_event.py
+++ b/addons/event/models/event_event.py
@@ -27,6 +27,7 @@ except ImportError:
 
 
 class EventType(models.Model):
+    _name = 'event.type'
     _description = 'Event Template'
     _order = 'sequence, id'
 

--- a/addons/event/models/event_mail_registration.py
+++ b/addons/event/models/event_mail_registration.py
@@ -9,6 +9,7 @@ _logger = logging.getLogger(__name__)
 
 
 class EventMailRegistration(models.Model):
+    _name = 'event.mail.registration'
     _description = 'Registration Mail Scheduler'
     _rec_name = 'scheduler_id'
     _order = 'scheduled_date DESC, id ASC'

--- a/addons/event/models/event_question.py
+++ b/addons/event/models/event_question.py
@@ -5,6 +5,7 @@ from odoo.exceptions import UserError
 
 
 class EventQuestion(models.Model):
+    _name = 'event.question'
     _rec_name = 'title'
     _order = 'sequence,id'
     _description = 'Event Question'

--- a/addons/event/models/event_registration.py
+++ b/addons/event/models/event_registration.py
@@ -11,6 +11,7 @@ _logger = logging.getLogger(__name__)
 
 
 class EventRegistration(models.Model):
+    _name = 'event.registration'
     _description = 'Event Registration'
     _inherit = ['mail.thread', 'mail.activity.mixin']
     _order = 'id desc'

--- a/addons/event/models/event_stage.py
+++ b/addons/event/models/event_stage.py
@@ -5,6 +5,7 @@ from odoo import fields, models
 
 
 class EventStage(models.Model):
+    _name = 'event.stage'
     _description = 'Event Stage'
     _order = 'sequence, name'
 

--- a/addons/event/models/event_tag.py
+++ b/addons/event/models/event_tag.py
@@ -7,6 +7,7 @@ from odoo import api, fields, models
 
 
 class EventTagCategory(models.Model):
+    _name = 'event.tag.category'
     _description = "Event Tag Category"
     _order = "sequence"
 
@@ -24,6 +25,7 @@ class EventTagCategory(models.Model):
 
 
 class EventTag(models.Model):
+    _name = 'event.tag'
     _description = "Event Tag"
     _order = "category_sequence, sequence, id"
 

--- a/addons/event/models/event_ticket.py
+++ b/addons/event/models/event_ticket.py
@@ -9,6 +9,7 @@ from odoo.tools.misc import formatLang
 
 
 class EventTypeTicket(models.Model):
+    _name = 'event.type.ticket'
     _description = 'Event Template Ticket'
     _order = 'sequence, name, id'
 

--- a/addons/event/models/mail_template.py
+++ b/addons/event/models/mail_template.py
@@ -6,7 +6,7 @@ from odoo.osv import expression
 
 
 class MailTemplate(models.Model):
-    _inherit = ['mail.template']
+    _inherit = 'mail.template'
 
     @api.model
     def _search_display_name(self, operator, value):

--- a/addons/event/models/res_config_settings.py
+++ b/addons/event/models/res_config_settings.py
@@ -8,7 +8,7 @@ from odoo import _, api, exceptions, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     def _default_use_google_maps_static_api(self):
         api_key = self.env['ir.config_parameter'].sudo().get_param('google_maps.signed_static_api_key')

--- a/addons/event/models/res_partner.py
+++ b/addons/event/models/res_partner.py
@@ -12,7 +12,7 @@ from odoo import api, fields, models
 
 
 class ResPartner(models.Model):
-    _inherit = ['res.partner']
+    _inherit = 'res.partner'
 
     event_count = fields.Integer(
         '# Events', compute='_compute_event_count', groups='event.group_event_registration_desk')
